### ISSUE TITLE
Failure to start app

### ISSUE THREAD

**Issue by osv** [sehrope/dokku-logging-supervisord#18]
I just tried simple example https://github.com/heroku/node-js-sample with Procfile
```
web: node index.js
```

I got lots of restarts of process 
```
tail -f /var/log/dokku/testjs/web.00.log
2014-12-23 15:52:21,643 INFO supervisord started with pid 175
2014-12-23 15:52:22,645 INFO success: web entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2014-12-23 15:52:22,647 INFO spawned: 'web' with pid 178
Generated supervisord.conf:
     1  [supervisord]
     2  logfile=/var/log/app/supervisor.log
     3  loglevel=info
     4  nodaemon=true
     5
     6  [program:web]
     7  command=/exec 'node index.js'
     8  process_name:%(program_name)s
     9  numprocs=1
    10  autostart=true
    11  autorestart=true
    12  stopsignal=QUIT
    13  redirect_stderr=true
    14  stdout_logfile=/var/log/app/%(program_name)s.%(process_num)02d.log
2014-12-23 15:52:22,788 CRIT Supervisor running as root (no user in config file)
2014-12-23 15:52:22,791 INFO supervisord started with pid 189
2014-12-23 15:52:23,793 INFO success: web entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2014-12-23 15:52:23,796 INFO spawned: 'web' with pid 192
Generated supervisord.conf:
     1  [supervisord]
     2  logfile=/var/log/app/supervisor.log
     3  loglevel=info
     4  nodaemon=true
     5
     6  [program:web]
     7  command=/exec 'node index.js'
     8  process_name:%(program_name)s
     9  numprocs=1
    10  autostart=true
    11  autorestart=true
    12  stopsignal=QUIT
    13  redirect_stderr=true
    14  stdout_logfile=/var/log/app/%(program_name)s.%(process_num)02d.log
2014-12-23 15:52:23,936 CRIT Supervisor running as root (no user in config file)
2014-12-23 15:52:23,940 INFO supervisord started with pid 203
```

````
tail -f /var/log/dokku/testjs/supervisor.log 
2014-12-23 15:52:25,089 CRIT Supervisor running as root (no user in config file)
2014-12-23 15:52:25,092 INFO supervisord started with pid 217
2014-12-23 15:52:26,095 INFO success: web entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2014-12-23 15:52:26,097 INFO spawned: 'web' with pid 220
2014-12-23 15:52:26,244 CRIT Supervisor running as root (no user in config file)
2014-12-23 15:52:26,248 INFO supervisord started with pid 231
2014-12-23 15:52:27,250 INFO success: web entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2014-12-23 15:52:27,251 INFO spawned: 'web' with pid 234
2014-12-23 15:52:27,400 CRIT Supervisor running as root (no user in config file)
2014-12-23 15:52:27,403 INFO supervisord started with pid 245
2014-12-23 15:52:28,404 INFO success: web entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2014-12-23 15:52:28,406 INFO spawned: 'web' with pid 248
2014-12-23 15:52:28,545 CRIT Supervisor running as root (no user in config file)
2014-12-23 15:52:28,549 INFO supervisord started with pid 259
2014-12-23 15:52:29,552 INFO success: web entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2014-12-23 15:52:29,555 INFO spawned: 'web' with pid 262
2014-12-23 15:52:29,712 CRIT Supervisor running as root (no user in config file)
2014-12-23 15:52:29,717 INFO supervisord started with pid 273
```
I use this (with hacked /build dir) and fork https://github.com/expa/dokku-logging-supervisord, both have same behaviors.

**Comment by michaelshobbs** [comment]
@sehrope in the latest buildstep release `/exec` does a couple things which I think are integral to dokku apps working correctly.

 - ensures the app files are owned by the non-root user 
 - sources the app's profile.d settings
 - runs the command as the non-root user

I think moving away from the standard execution model can/will cause more issues down the road.

**Comment by sehrope** [comment]
@michaelshobbs Yeah exactly. That's why I don't want to touch it.

If the environment is already setup by the parent that is calling supervisord then there is no need to re-initialize the environment before starting the child processes (i.e. those listed in the Procfile). They can be executed directly.

**Comment by osv** [comment]
however, have a copy exec as exec_svd good too :) but, one extra docker commit, that make litlebit slower

**Comment by michaelshobbs** [comment]
@sehrope good point. I'd like to test out some of my envs with this modification. Probably tomorrow. I'll report back any oddities.

**Comment by michaelshobbs** [comment]
@osv why do we need a copy of it?

**Comment by osv** [comment]
Because of Python, I don't test python buildpack, but I think supervisord have problem with them because of setting LIBRARY_PATH. As I remeber your fork unset this env before running supervisord.
You use own buildpack that have no "ln -s" but "cp /start /exec" :) so you don't need copy, like me, that want use progrium/buildstep
https://github.com/heroku/heroku-buildpack-python/blob/master/bin/compile#L181

**Comment by michaelshobbs** [comment]
@osv I snagged some of your changes from expa#1 and included them PR #20

**Comment by michaelshobbs** [comment]
Scratch my previous statement @osv. I actually think we shouldn't copy /exec to another location. Instead we should unlink it from /start and cp /start to /exec. Otherwise we end up with /exec being our supervisor invocation and that may break other plugins that rely on /exec. See #20

**Comment by osv** [comment]
Agree

**Comment by sehrope** [comment]
Ok I think I got everything straightened out now. Check out the [devel](https://github.com/sehrope/dokku-logging-supervisord/tree/devel) branch.

I combined the open PRs manually so it's a separate pair of commits. The first is a pretty small one to improve the helpers a bit further. The second is the real one that actually fixes things to work with the new buildstep. 

Seems like everything is working fine now. Let me know if either of you have issues with it (or the approach). If all is well, I'll merge into master.

**Comment by michaelshobbs** [comment]
I should have some time to bang away at this a bit today

**Comment by michaelshobbs** [comment]
Tested and commented.

**Comment by sehrope** [comment]
I merged the fixes for this into master.
